### DISCUSSION TITLE
Custom eslint plugin

### DISCUSSION THREAD

**Discussion by stafyniaksacha** [DamianOsipiuk/vue-query#77] [Ideas]
Hello!

This implementation is really great ! Thanks for the work :)
As vue does not have the same system as react hooks, we do not benefit of some of those features with linters (I mostly think about [`react-hooks/exhaustive-deps`](https://github.com/facebook/react/tree/main/packages/eslint-plugin-react-hooks) rule and the similarity it has with [query keys and functions variables](https://vue-query.vercel.app/#/guides/query-keys?id=if-your-query-function-depends-on-a-variable-include-it-in-your-query-key))

Do you think it can be relevant to have this kind of eslint plugin for vue-query ?

**Comment by DamianOsipiuk** [comment]
Hi,

Since the Vue reactivity system works a little differently than React, you actually do not need to track reactive props, as it's done automatically by Vue.

As for other rules, I see that there is a common theme of people passing `ref` or `computed` directly to the reactive array, which breaks reactivity for them. (https://v3.vuejs.org/guide/reactivity-fundamentals.html#access-in-reactive-objects - ref unwrapping paragraph)
Additionally, reactive arrays have other quirks that could be covered by the eslint rule, like adding new items to the array via `Vue.set`.

So, yes such rule could be beneficial. But I would say it's a general vue reactivity rule, not specifically vue-query related.

https://v3.vuejs.org/guide/change-detection.html#for-arrays
